js/mexc3.js

```diff
@@ -4002,6 +4025,120 @@ module.exports = class mexc3 extends Exchange {
+        for (let i = 0; i < response.length; i++) {
+            const item = response[i];
+            const symbol = this.safeString(item, 'symbol');
+            const data = this.safeValue (item, 'data');
+            for (let j = 0; j < data.length; j++) {
+                const coin = data[j];
+                const currency = this.safeString (coin, codeKey);
+                const code = this.safeCurrencyCode (currency);
+                const borrowRate = this.parseBorrowRate (coin);
+                result[code] = borrowRate;
+            }
+        }
+        return result;
+    }
+
+    parseBorrowRate (info, currency = undefined) {
+        //
+        //     {
+        //         "coin": "BTC",
+        //         "hourInterest": "0.00026125",
+        //         "borrowLimit": "270"
+        //     }
+        //
+        const coin = this.safeString (info, 'coin');
+        return {
+            'currency': this.safeCurrencyCode (coin),
+            'rate': this.safeNumber (info, 'hourInterest'),
+            'period': undefined,
+            'timestamp': undefined,
+            'datetime': undefined,
+            'info': info,
+        };
+    }
+
+    async fetchBorrowRates (params = {}) {
+        await this.loadMarkets ();
+        const response = await this.spotPrivateGetMarginIsolatedMarginData (params);
+        //
+        //     [
+        //         [
+        //             {
+        //                 "symbol": "BTCUSDT",
+        //                 "leverage": "10",
+        //                 "data": [
+        //                     {
+        //                         "coin": "BTC",
+        //                         "hourInterest": "0.00026125",
+        //                         "borrowLimit": "270"
+        //                     }, {
+        //                         "coin": "USDT",
+        //                         "hourInterest": "0.000475",
+        //                         "borrowLimit": "2100000"
+        //                     }
+        //                 ]
+        //             }
+        //         ]
+        //     ]
+        //
+        return this.parseBorrowRates (response, 'coin');
+    }
+
+    parseBorrowInterest (info, market = undefined) {
+        const coin = this.safeString (info, 'asset');
+        const timestamp = this.safeInteger (info, 'timestamp');
+        const datetime = this.iso8601 (timestamp);
+        const remainInterest = this.safeString (info, 'remainInterest');
+        const repayInterest = this.safeString (info, 'repayInterest');
+        const remainAmount = this.safeString (info, 'remainAmount');
+        const repayAmount = this.safeString (info, 'repayAmount');
+        return {
+            'account': 'isolated',
+            'symbol': this.safeString (info, 'symbol'),
+            'marginMode': 'isolated',
+            'currency': this.safeCurrencyCode (coin),
+            'interest': Precise.stringAdd(remainInterest, repayInterest),
```

Comment:
When returning these objects directly I think we should be using "number" type for numbers. If we're sending it to some parser like safeOrder, safeTrade, etc then it is ok and advisable to send strings